ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QuerySubscriber.java

```diff
@@ -93,9 +94,19 @@ private void checkMakeRequest() {
 
   @Override
   public void handleError(final Throwable t) {
-    log.error("Error in processing query", t);
-    final KsqlErrorMessage errorResponse = new KsqlErrorMessage(ERROR_CODE_SERVER_ERROR,
-        "Error in processing query. Check server logs for details.");
+    final StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append(t);
+    for (Throwable s: t.getSuppressed()) {
+      if (s instanceof NotUpToBoundException) {
+        stringBuilder.append(" Failed to get value from materialized table, reason: "
+                                 + "NOT_UP_TO_BOUND");
+      } else {
+        stringBuilder.append(s.getMessage());
+      }
+    }
+    final KsqlErrorMessage errorResponse = new KsqlErrorMessage(
+        ERROR_CODE_SERVER_ERROR, stringBuilder.toString());
+    log.error("Error in processing query {}", stringBuilder);
```

Comment:
Hmm, I am conflicted about this. I thought it would be helpful for the user to know what nodes we tried and why they failed so that they can "debug" or at least understand why routing failed. I feel we get a lot of questions/complaints about pull queries failing on standbys and the users could not understand why.